runtime/js/30_fs.js

```diff
@@ -148,46 +154,120 @@
     path,
     options = {},
   ) {
-    ops.op_remove_sync({
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    ops.op_remove_sync(
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   async function remove(
     path,
     options = {},
   ) {
-    await core.opAsync("op_remove_async", {
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    await core.opAsync(
+      "op_remove_async",
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   function renameSync(oldpath, newpath) {
-    ops.op_rename_sync({
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    ops.op_rename_sync(
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
   async function rename(oldpath, newpath) {
-    await core.opAsync("op_rename_async", {
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    await core.opAsync(
+      "op_rename_async",
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
+  // Extract the FsStat object from the encoded buffer.
+  // See `runtime/ops/fs.rs` for the encoder.
+  //
+  // This is not a general purpose decoder. There are 4 types:
+  //
+  // 1. date
+  //  offset += 4
+  //  1/0 | extra padding | high u32 | low u32
+  //  if date[0] == 1, new Date(u64) else null
+  //
+  // 2. bool
+  //  offset += 2
+  //  1/0 | extra padding
+  //
+  // 3. u64
+  //  offset += 2
+  //  high u32 | low u32
+  //
+  // 4. ?u64 converts a zero u64 value to JS null on Windows.
+  function createByteStruct(types) {
+    // types can be "date", "bool" or "u64".
+    // `?` prefix means optional on windows.
+    let offset = 0;
+    let str =
+      'const unix = Deno.build.os === "darwin" || Deno.build.os === "linux"; return {';
+    for (let [name, type] of ObjectEntries(types)) {
+      const optional = type.startsWith("?");
+      if (optional) type = type.slice(1);
+
+      if (type == "u64") {
+        if (!optional) {
+          str += `${name}: view[${offset}] + view[${offset + 1}] * 2**32,`;
+        } else {
+          str += `${name}: (unix ? (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) : (view[${offset}] + view[${
+            offset + 1
+          }] * 2**32) || null),`;
+        }
+      } else if (type == "date") {
+        str += `${name}: view[${offset}] === 0 ? null : new Date(view[${
+          offset + 2
+        }] + view[${offset + 3}] * 2**32),`;
+        offset += 2;
+      } else {
+        str += `${name}: !!(view[${offset}] + view[${offset + 1}] * 2**32),`;
+      }
+      offset += 2;
+    }
+    str += "};";
+    return [new Function("view", str), new Uint32Array(offset)];
```

Comment:
Is there a way we could not do an eval like that?

Reply:
yes, by writing it manually which is going to be uglier than this. this is top level only btw

Reply:
Yeah, only reason I am fine with this is because it executes during snapshot. Maybe add a comment explaining that?